### PR TITLE
Rename MeterRegistry -> MeterProvider

### PR DESCRIPTION
**Pull request by thisthat**
This addresses #794 

Specification: [link](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#obtaining-a-meter)

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/metrics/MeterProvider.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
  * @since 0.1.0
  */
 @ThreadSafe
-public interface MeterRegistry {
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/metrics/MeterProvider.java]: Please comment that this is called `Provider` for consistency with other languages but it is not loaded using reflection (ServiceLoader)

File: api/src/test/java/io/opentelemetry/metrics/DefaultMeterTest.java [hunk 2]
```diff
@@ -56,6 +56,11 @@ public void noopAddMeasureLong_NullName() {
     defaultMeter.longMeasureBuilder(null);
   }
 
+  @Test
+  public void noopAddMeasureLong_EmptyName() {
```
Comment (bogdandrutu) [inline api/src/test/java/io/opentelemetry/metrics/DefaultMeterTest.java]: Add an issue about this, I don't think we should accept empty name, but we can address this separately.

Comment (thisthat) [reply api/src/test/java/io/opentelemetry/metrics/DefaultMeterTest.java]: I was about to create the issue :wink:

**Comment by bogdandrutu** [comment]
Same change is required for trace.

**Comment by thisthat** [comment]
> Same change is required for trace.

Yes, my plan is to split up the renaming in multiple PRs

**Pull request merged**